packages/dialect-react-ui/components/NotificationsButton/index.tsx

```diff
@@ -69,10 +70,41 @@ function WrappedNotificationsButton(
   const wrapperRef = useRef(null);
   const bellRef = useRef(null);
   const [open, setOpen] = useState(false);
+  const [hasNewMessages, setHasNewMessage] = useState(false);
+
   useOutsideAlerter(wrapperRef, bellRef, setOpen);
   const { setWallet, setNetwork, setRpcUrl } = useApi();
   const isWalletConnected = connected(props.wallet);
 
+  const { messages } = useDialect();
+
+  useEffect(() => {
+    if (!window) return;
+
+    const lastMsgTimestampStorage = window.localStorage.getItem('lastMsg');
```

Comment:
Also, I think we need to also abstract the logic of getting the `lastMsg` timestamp, so that later we can just change it to server fetching?

Also-also, given that we have one localStorage, but potentially many windows related to dialect, would it make sense to track which thread this `lastMsg` is related to? What I'm saying, I think it's worth changing the storage to a map:
```
{
  [threadId: string]: string; // timestamp
}
```
That way if we decide to scale to multi-thread tracking, it would be easy to add this to our localStorage.

And final also-also-also, I think we need more control over localStorage in case of any bugs or structure changes, since we're essentially using it as our DB. Please consider adding migrations functionality to the structure (e.g. https://github.com/ragnarstolsmark/localstorage-migrator), so that we can update the structure of the localstorage to our needs, in case we update it in any way. This will give us control over something that we have no access to. Please feel free to challenge this.

Reply:
This make perfect sense, fix it, please review it